### PR TITLE
(SIMP-5491) Harden all ssh_host_ keys

### PR DESCRIPTION
**Pull request by ayohrling**
This adds a couple of keys to the server.pp file to set to 0600 and 0644
for private and public, respectively.

Also, enables the permissions check within the inspec acceptance testing
for proper validation.

SIMP-5491 #closes

### CODE REVIEW

**Comment by ayohrling** [comment]
Acceptance: https://gitlab.com/simp/pupmod-simp-ssh/pipelines/33647806

**Comment by trevor-vaughan** [comment]
@ayohrling There are already facts for the SSH keys on the system (just not their paths). Do you think we can safely extrapolate from that?

Run `facter | grep ssh` to see them.

**Comment by ayohrling** [comment]
Valid acceptance tests will be dependent on https://github.com/simp/rubygem-simp-rspec-puppet-facts/pull/15 leaving as WIP until merged

**Comment by ayohrling** [comment]
⏳ Acceptance: https://gitlab.com/simp/pupmod-simp-ssh/pipelines/34475711

**Comment by trevor-vaughan** [comment]
:heavy_check_mark: https://gitlab.com/simp/pupmod-simp-ssh/pipelines/34475711

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -14,7 +14,6 @@
           it { is_expected.to create_class('ssh') }
           it { is_expected.to compile.with_all_deps }
           it { is_expected.to create_file('/etc/ssh') }
-          it { is_expected.to create_file('/etc/ssh/ssh_host_key') }
```
Comment (trevor-vaughan) [inline spec/classes/init_spec.rb]: This should get updated with the keys with the new facts.

Comment (trevor-vaughan) [reply spec/classes/init_spec.rb]: I'll take care of this

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-ssh/pipelines

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-ssh/pipelines/34485557

**Pull request merged**